### PR TITLE
PLANET-6196: Add credits to image caption in Slider Gallery

### PR DESCRIPTION
**Pull request by lithrel**
Ref: https://jira.greenpeace.org/browse/PLANET-6196

Image credits are not displayed in the Gallery block for the Slider version. (other versions don't display any text)

## Fix 

Exporting `getImageTitle()` function to use for frontend/backend and Lightbox data.

## Test

- Add a gallery to a post/page
- Use images that have a completed `credits` field
- On the Slider version of the gallery, credits should appear at the end of the image caption, on backend and frontend

Screenshot from 2021-06-16 12-06-28

### CODE REVIEW

File: assets/src/blocks/Gallery/getImageTitle.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const getImageTitle = (image) => {
+  const caption = image.caption || '';
+  const credits = image.credits && !caption.includes(image.credits)
+    ? (image.credits.includes('©') ? image.credits : `© ${image.credits}`) : '';
```
Comment (Inwerpsel) [inline assets/src/blocks/Gallery/getImageTitle.js:4]: There's not many cases where a double ternary is readable, and I'm afraid this is not one of them :stuck_out_tongue:

Comment (Inwerpsel) [reply assets/src/blocks/Gallery/getImageTitle.js:4]: You could instead do
```js
const credits = !image.credits || caption.includes(image.credits) ? '' : `© ${image.credits.replace(/^©\s*)}`;

Comment (Inwerpsel) [reply assets/src/blocks/Gallery/getImageTitle.js]: Maybe an additional const helps too
```js
const dontAddCredits = !image.credits || caption.includes(image.credits);
const credits = dontAddCredits ? '' : `© ${image.credits.replace(/^©\s*)}`;

File: assets/src/blocks/Gallery/getImageTitle.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export const getImageTitle = (image) => {
+  const caption = image.caption || '';
+  const credits = getCredits(image, caption);
+  return `${caption}  ${credits}`.trim();
+};
+
+const getCredits = (image, caption) => {
```
Comment (Inwerpsel) [inline assets/src/blocks/Gallery/getImageTitle.js:7]: That's another way to make it readable :+1: If you want you can include parts of the suggestion I commented right before you pushed this :smile: https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/613#discussion_r652566019

Comment (Inwerpsel) [reply assets/src/blocks/Gallery/getImageTitle.js:7]: Though those may not be needed now it's isolated, either way works for me :+1:

File: assets/src/blocks/Gallery/GalleryCarousel.js [hunk 3]
```diff
@@ -107,7 +108,7 @@ export const GalleryCarousel = ({ images, onImageClick }) => {
             {(image.caption || image.credits) && (
               <div className="carousel-caption">
                 <p>
-                  {image.caption || image.credits}
+                  {getImageTitle(image)}
```
Comment (Inwerpsel) [inline assets/src/blocks/Gallery/GalleryCarousel.js:111]: Maybe `getCaptionWithCredits` is a bit more accurate?

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [tavros](https://www-dev.greenpeace.org/test-tavros)

:watch: 2021.06.16 10:49:10

**Review by Inwerpsel — APPROVED** [review]

**Pull request merged**